### DISCUSSION TITLE
Fairlearn website structure

### DISCUSSION THREAD

**Discussion by MiroDudik** [fairlearn/fairlearn#673] [General]
There are many issues with our current website, so this discussion is to figure out the content structure that we'd like to see. I'm starting with a navigation map, based on the items in the navigation bar:

* Fairlearn -> Landing page
  * [blocks to be specified]
* Getting started [first time user is the target]
  * How to install
  * Overview of the pieces of the project highlighting socio-technical nature, including use cases, contributor guide, what we mean by fairness [but some of this might actually just better fit on the landing page?]
  * Short sections with very small examples; however, we should flag socio-technical issues/limitations
* Learn
  * User guide [course on fairlearn]
  * Code examples [.py that does not go into docstrings]
  * Case studies [.ipynb / .rst]
  * Further resources
* API Docs [from docstrings]
* Contribute
* Community
* FAQ
* About us
* [GitHub icon] -> GitHub repository

Thoughts? Comments? Questions?

**Comment by romanlutz** [comment]
# Community & contributor guide discussion

I had "community" and "contributor guide" (the current name, although I like "contribute" as well!) originally separate, but merged them shortly after since they were cross-referencing each other all the time. Typically, the community is the set of people who contribute so I'm still for keeping that as a single point. Note that contributing does not necessarily mean writing code. There are many ways people are already contributing without writing code (for more see [ways to contribute](https://fairlearn.github.io/v0.5.0/contributor_guide/ways_to_contribute.html))

[Please respond to this particular thread if you have thoughts on community & contributor guide/contribute. For discussions on other parts just start a new thread below]

**Reply by MiroDudik** [reply]
I like the idea! How about something like this:

* Community
  * Join our community [gitter / stack overflow / slack? / weekly community calls]
  * Code of conduct [should we put it here or under About?]
  * Ways to contribute [Contributor guide]

**Reply by romanlutz** [reply]
That seems like a subset of what we currently have there. Any reason for removing other points?
https://fairlearn.github.io/v0.5.0/contributor_guide/index.html

**Reply by nessamilan** [reply]
I think the issue is that the top level label as "contributor guide" is not intuitive as a starting point to _learn_ about the many ways one can contribute Fairlearn. "Guide" feels like it speaks more to a user who's already made the decision to contribute to the code base (I could be biased since I normally see this label in developer guides :)). This _may_ deter those who want to contribute in other ways. Dropping "guide" may help this section feel more inclusive. I like the way TensorFlow [breaks down this section](https://www.tensorflow.org/community/contribute).

"Community" refers more to the higher level social dynamics of engagement. This can range from chats/conversations, real-time content consumption (i.e. passive engagement with threads on all platforms) and public workshops to contributing in a meaningful way (i.e. including code, content, graphics/data visualizations, etc.).  

The question is, In the flow of a "new to Fairlearn" user journey, what's most intuitive for our users? There are good arguments for "Contribute" as a tier 1 link or subsection of "Community". My hunch is that data scientists newer to Fairlearn are interested in the "latest" and experimenting, while returning/familiar/confident users may want to contribute and therefore begin this journey quickly. 

**Guide entry point ideas:**
- Tier 1 navigation, simplify to "contribute" to be more inclusive
- Subsection of "Community"
- Tier 1 + module on "Community" landing page that points to tier 1 section
- Tier 1 + module on homepage + module in community section (may seem redundant, but we cannot predict where/when users will always make the decision to learn about or actually contribute)

@MiroDudik I keep going back and forth about Code of Conduct. When present, many sites have this in Community, while others have it in "About", or simply just a footer link to it's own "free-floating" page... Often I've seen it repeated in GitHub when necessary.

**Reply by hildeweerts** [reply]
I like simplifying to contribute! Maybe a fifth option is Subsection of "Community" + module on homepage?

Re. code of conduct: I don't have a very strong opinion, but I would personally put it under community. The definition of code of conduct is really the norms/rules/etc. we want to uphold within the entire community, not just within the "official" organization. Also, we currently literally refer to "GitHub Community Guidelines", which gives a hint about where it should go I think.

**Reply by romanlutz** [reply]
Did we ever settle on any of those options? In the latest discussion #685 it seems like there's still Community and Contribute as separate items. The current equivalent of "community" would be half of this page: https://fairlearn.github.io/v0.5.0/contributor_guide/index.html which is basically two paragraphs. If we plan on significantly expanding it I don't mind having it as a separate top-level item, but otherwise it seems a bit much (?)

**Comment by MiroDudik** [comment]
# Designs that we like

Please put in replies some of the existing designs that we can use for inspiration and maybe also explain what you like or dislike about them.

**Reply by romanlutz** [reply]
I'll start with a few!

1. [scikit-learn](https://scikit-learn.org/stable/)
  - I like that I can see EVERYTHING from the landing page, which is very different from ours. With ours, I basically always need an additional click to get where I want to go since I need to navigate away from the landing page first. This landing page is designed as a place from where you can navigate anywhere you want immediately. The downside of that is that a lot is going on there, of course.
  - It's unclear to me how user guide is different from tutorials.
2. [pandas](https://pandas.pydata.org/)
  - Really dislike that when you click on "documentation" you end up on a different page and the top bar is completely different. We kind of have the same experience because the landing page is different.
  - We are using the pydata-sphinx-theme, so if there's anything they have that we like it's easy to get that ;-)
3. [pytorch](https://pytorch.org/)
  - Looks great overall to me and it's consistent across all pages with the same bar at the top.
  - no separate user guide/API docs, it's just all a single "Docs" section. Example: https://pytorch.org/docs/stable/tensors.html I kind of like not having to go to several different pages to find all the info.
  - I like the "Resources" tab as a catch-all for all kinds of categories: 
![image](https://user-images.githubusercontent.com/10245648/104061393-560fcc00-51ad-11eb-91f7-aa369bb293e0.png) If we run out of space in the bar that could replace several individual sections there that are probably more relevant to contributors rather than users.

**Reply by hildeweerts** [reply]
I agree with all your comments, @romanlutz!

I think it wouldn't be too difficult to design a landing page that's a bit less overwhelming yet still contains all navigation. I don't think I've ever used any of the links to the user guide on the landing page, though :')

I think something like pytorch's "Resources" tab would be great for our "Learn", because it allows for a bit more explanation on what's actually in there (although that's actually not available on the mobile version, but I guess most people will be browsing from their computer anyways).

**Reply by nessamilan** [reply]
Agree with many of @romanlutz 's comments.

I like the simplicity of how **TensorFlow** organizes their site, including the specific labels they use:

- They break out the highest priority documentation into the tier 1, i.e. Install, Learn, API
- Community includes "Contributions" which I think is an intuitive way to group how the community can "get involved'. I'm not sure about breaking this out into its own tab as mapped in @MiroDudik 's discussion starter. Adding more comments to Contributor & community guide discussion thread.

<img width="843" alt="Screen Shot 2021-01-11 at 12 04 53 PM" src="https://user-images.githubusercontent.com/6819397/104232825-bbf69080-5405-11eb-9f6a-692b0698b039.png">

**Comment by hildeweerts** [comment]
# Getting Started
To avoid reiterating the landing page, we can think carefully about what would be the right type of content re. sociotechnical context for a first time user. For example, I think understanding the different types of harm & how your problem formulation influences the  impact of your system (i.e. abstraction traps) would be a first priority, as opposed to very industry-specific use cases. 

What do people think?
